cmd/homi/setup/cmd.go

```diff
@@ -1132,14 +1134,31 @@ func removeSpacesAndLines(b []byte) string {
 func homiFlagsFromYaml(ctx *cli.Context) error {
 	filePath := ctx.String(homiYamlFlag.Name)
 	if filePath != "" {
-		if err := altsrc.InitInputSourceWithContext(SetupCommand.Flags, altsrc.NewYamlSourceFromFlagFunc(homiYamlFlag.Name))(ctx); err != nil {
+		if err := altsrc.InitInputSourceWithContext(HomiFlags, altsrc.NewYamlSourceFromFlagFunc(homiYamlFlag.Name))(ctx); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+// In urfave/cli/v2 flags must come before args
+func CheckArgs(ctx *cli.Context) error {
+	args := ctx.Args().Slice()
+	for i, arg := range args {
+		if strings.HasPrefix(arg, "--") {
+			if i > 1 {
+				return fmt.Errorf("flag %s must come before arguments", arg)
```

Comment:
Can't we just move the misplaced flags to the right of the arguments? Is there any huddle to do it?

Reply:
@yoomee1313 It's quite complicated since we actually get the flag data from context.
The context is in the library area.